docs/DeveloperGuide.md

```diff
@@ -1,16 +1,38 @@
 # Developer Guide
 
 ## Design & implementation
+<!-- @@author iamchenjiajun -->
 ### Architecture
 
+##### Architecture Diagram
 Below is an architecture diagram of termiNus.
 
 ![ArchitectureDiagram](./images/ArchitectureDiagram.png)
 
+- `Duke` is the main object of the program and handles all the logic and models related to the program.
+- `Ui` is the main object that provides an interface between `Duke` and the user.
+- `Command` represents a command that is provided by the user. `Ui` reads the command and it is sent to `Parser` to create a new `Command` object.
+- `Duke` executes the command and shows the output to the user using `Ui`.
+- `Command` object modifies the state of `ItemList`, which consists of multiple lists for different types of `Items`.
+- `Storage` takes the state of `ItemList` and stores it to file.
```

Comment:
Good flow! Perhaps storage should be accessible only by ItemList, since storage should only be changed when ItemList changes.